contracts/src/common/DedicatedResolver.sol

```diff
@@ -0,0 +1,300 @@
+    function initialize(address owner, bool _wildcard, address findResolverImpl) public initializer {
+        __Ownable_init(owner);
+        wildcard = _wildcard;
+        emit NewDediciatedResolver(owner, _wildcard);
+        if (findResolverImpl != address(0)) {
+            _setInterface(type(IFindResolver).interfaceId, findResolverImpl);
+        }
+    }
+
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165) returns (bool) {
+        return interfaceId == type(IExtendedResolver).interfaceId || interfaceId == type(IDedicatedResolver).interfaceId
+            || interfaceId == type(IMulticallable).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /// @notice Get the `IFindResolver` implementation.
+    ///         Defaults to the proxy implementation if unset.
+    function findResolverImplementation() public view returns (address impl) {
+        impl = interfaceImplementer(NODE_ANY, type(IFindResolver).interfaceId);
+        if (impl == address(0)) {
+            address base = ERC1967Utils.getImplementation();
+            if (base != address(0)) {
+                impl = DedicatedResolver(base).findResolverImplementation();
+            }
+        }
+    }
+
+    /// @notice Set address for the coin type.
+    ///         If coin type is EVM, require exactly 0 or 20 bytes and replace empty 20 bytes with 0 bytes.
+    /// @param coinType The coin type.
+    /// @param addressBytes The address to set.
+    function setAddr(uint256 coinType, bytes memory addressBytes) external onlyOwner {
+        if (addressBytes.length != 0 && ENSIP19.isEVMCoinType(coinType)) {
+            if (addressBytes.length != 20) {
+                revert InvalidEVMAddress(addressBytes);
+            } else if (bytes20(addressBytes) == bytes20(0)) {
+                addressBytes = "";
+            }
+        }
+        _addresses[coinType] = addressBytes;
+        emit AddressChanged(NODE_ANY, coinType, addressBytes);
+        if (coinType == COIN_TYPE_ETH) {
+            emit AddrChanged(NODE_ANY, address(bytes20(addressBytes)));
+        }
+    }
+
+    /// @notice Get the address for coin type.
+    ///         If coin type is EVM and empty, defaults to `addr(EVM_BIT)`.
+    /// @param coinType The coin type.
+    /// @return addressBytes The address for the coin type.
+    function addr(bytes32, uint256 coinType) public view returns (bytes memory addressBytes) {
+        addressBytes = _addresses[coinType];
+        if (addressBytes.length == 0 && ENSIP19.chainFromCoinType(coinType) > 0) {
+            addressBytes = _addresses[EVM_BIT];
+        }
+    }
+
+    /// @notice Get `addr(60)` as `address`.
+    /// @return addr_ The address for coin type 60.
+    function addr(bytes32) public view returns (address payable addr_) {
+        addr_ = payable(address(bytes20(addr(NODE_ANY, COIN_TYPE_ETH))));
```

Comment:
By convention, use `return` instead of named return values, unless it'd be clearer (due to control flow) to do otherwise.